File: 1_BIMESTRE/ATIVIDADES/DAS2_UN3_ARQUITETURA_LIMPA/REFATORACAO/Compra/SpaghettiCompraController.java
import java.util.HashMap;
import java.util.Map;

public class SpaghettiCompraController {

    public Map<String, Object> calcularTotal(Map<String, Object> request) {
        Map<String, Object> response = new HashMap<>();

        try {
            Double valor = (Double) request.get("valor");
            String tipoCliente = (String) request.get("tipo_cliente"); // NORMAL | VIP

            if (valor == null || valor <= 0) {
                response.put("erro", "Valor inválido");
                response.put("status_code", 400);
                return response;
            }

            double total = valor;

            // desconto VIP
            if ("VIP".equals(tipoCliente)) {
                total = total * 0.9;
            }

            // desconto adicional
            if (valor > 100) {
                total = total * 0.95;
            }

            // valor mínimo
            if (total < 10) {
                total = 10;
            }

            response.put("valor_original", valor);
            response.put("tipo_cliente", tipoCliente);
            response.put("valor_final", Math.round(total * 100.0) / 100.0);
            response.put("status_code", 200);

            return response;

        } catch (Exception e) {
            response.put("erro", e.getMessage());
            response.put("status_code", 500);
            return response;
        }
    }
}

File: spaghetti_compra_controller.py
# compra_controller.py

class CompraController:

    def calcular_total(self, request):
        try:
            valor = request.get("valor")
            tipo_cliente = request.get("tipo_cliente")  # NORMAL | VIP

            if valor is None or valor <= 0:
                return {"erro": "Valor inválido", "status_code": 400}

            total = valor

            # desconto VIP
            if tipo_cliente == "VIP":
                total = total * 0.9

            # desconto adicional
            if valor > 100:
                total = total * 0.95

            # valor mínimo
            if total < 10:
                total = 10

            return {
                "valor_original": valor,
                "tipo_cliente": tipo_cliente,
                "valor_final": round(total, 2),
                "status_code": 200
            }

        except Exception as e:
            return {
                "erro": str(e),
                "status_code": 500
            }